refactor: Update PhotonPoseEstimator usage for new vendor library

ylukin opened the pull request:
Update Vision subsystem to use new PhotonPoseEstimator constructor and update() methods following PhotonVision API changes.

ylukin committed:
refactor: Update PhotonPoseEstimator usage for new vendor library

Update Vision subsystem to use new PhotonPoseEstimator constructor and update() methods following PhotonVision API changes.

ylukin commented:
fixes #1

Eli-Luk committed:
Got Steer Working

Steering works with 2025 REVLIB. Before we integrate Krakens we need to work on the superclass. Refer to this commit to find out what was commented out temporarily.

SoapyLiquids committed:
Started Kraken Integration

Created two new files and interfaces for the purpose of creating parallel methods and code for both Kraken and Neo motor controllers

skajacore reviewed (approved):
Seems reasonable. I haven't reviewed the documentation yet, but seems they just shifted the camera input from the pose estimator setup to the update function. We should probably look into why they did it as maybe there's a benefit in the new strategy that we're not taking advantage of.

SoapyLiquids committed:
Creation of Motor Class

Deleted the two old classes and created a singular Motor class that will be used to globalize all motor functions for all future use

SoapyLiquids committed:
Added methods to Motor

Added the remaining methods from SwerveModule lines 108 - 118 into Motor.java for Sparkmax, and added some functionality for TalonFX

ylukin commented:
I think this is the commit that changed the constructor method: [photonvision commit 67463a0](https://github.com/PhotonVision/photonvision/commit/67463a020a358b6bc4c62b49dbeb715682887477#diff-57934f2b05d27fb4c9e9977f789b03174884d323f86821e937b8976088a63f01)

I can't find anything related to this change in their docs...

skajacore commented:
I'll take a look through it. I also noticed there was a change that deprecated the method to get the pipeline latency, but the docs still reference it. We have to figure out how to get the processed data latency.

I found some documentation that said getting the image timestamp was the better than using the pipeline latency, but if the Rio FPGA clock isn't synchronized with the mini PC, that would cause issues. Not sure if that's already handled somehow.

Eli-Luk committed:
Continued work on Motor class

Updated PID, speed, and position methods for the Kraken in Motor class.

SoapyLiquids committed:
Woohoo more motor class

Added more methods for the talon motor in Motor.java

SoapyLiquids committed:
Corrected All References to Motors

Changed all previous references to SparkMax motors to the new Motor.java class. The setup commands are commented out, and will need to be moved to motor.java, but as of now the code should compile and run motors.

SoapyLiquids committed:
It Brokey

It doesnt work :(
Somehow of all things the Translation2d class is erroring...

SoapyLiquids committed:
Rewrote WaitUntil Commands

Brought the waitUntil Commands from the SwerveModule class and changed the names of the variables to match the new names from Motor class. Added definitions from SwerveModule class to fix the variables that were still erroring.

Eli-Luk committed:
Fixed String Logic

Quick fix to how the code detects whether a neo is a drive or steer motor.

SoapyLiquids committed:
Added Motor Setup Check

Added logic to motor class to check if the motor setup is done.

SoapyLiquids committed:
Steer Motors Work!

Updated the setup command sequence to drive the neo motors and fixed the deadband error where Rotation2D values of x and y are 0.

SoapyLiquids committed:
Fixed Calibration logic & Krakens move

Fixed Calibration logic so that the steering offset is set and not equal to zero. Wheels were calibrated, PID gains need to be adjusted to stop Krakens from jamming.

SoapyLiquids committed:
Kraken Drive Works

Adjusted PID gains and debugged to get Krakens working. Uncommented code that tells motor to actually drive and added code in Motor to apply config after setupmotor is done in pid.

ylukin committed:
some random fixes

publish steeringoffset to network tables, also fix velocityconversionfactor to call the correct method

skajacore committed:
"bug fixes and performance improvements"

Refactored lambda for setting control gains over network tables, which made explicit methods in Motor.java for each value.
Used CommandSequence to make sure gains get pushed out to motors without issue, when updated.
Updated Gyro simulation to use the Pigeon2SimState.addYaw() to eliminate warnings about using the hardware API for simulation.
Switched to non-static method of SwerveModuleState.optimize() to get rid of deprecated code use.
Added success status feedback to setSpeed and setPosition for future fault detection use.

skajacore committed:
Warning Cleanup

Eli-Luk committed:
Removed warnings & fixed init math

Removed some more import warning and fixed the angle initialization math. Updated libraries.

Eli-Luk committed:
Tweaked PID gains & fixed velocity math.

Tweaked the PID gains so that they are looking about right. Fixed velocity math so that the units align.

skajacore commented:
Rebasing main into this made quite a mess of the commit history. I'll make a new branch and PR.
